blueprints/multi-interface-servers.md

```diff
@@ -0,0 +1,829 @@
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+# Additional Server Interfaces
+
+## Problem Description
+Multiple network interfaces may be used by cache servers to service requests.
+However, these cannot be expressed as parts of the current "server" object.
+Specifically, assuming each interface has its own IP address, then each would
+need to be treated as a separate cache server. This means that getting clients
+to use one when the other is unavailable requires manual intervention.
+
+## Proposed Change
+Instead, cache servers should be configurable with numerous network interfaces
+monitored by Traffic Monitor that are all associated with the server, each
+having its own associated network information e.g. IP Address, but Traffic
+Router should only be aware of the "main" interface on which the cache server
+listens for connections.
+
+The specific use-case outlined above, obviously, entails some management be done
+on the cache servers to actually support multiple interfaces answering to the
+same "main" IP address and this should be considered outside of
+<abbr title="Apache Traffic Control">ATC</abbr>'s scope of operation.
+
+## Data Model Impact
+The proposed structure of each of these interfaces is given below as a pair of
+TypeScript<sup>[1](#typescript)</sup> interfaces:
+```typescript
+interface IPAddress {
+	/**
+	 * The actual IP (v4 or v6) address which is used by an interface.
+	 * If it is an IPv6 address
+	 */
+	address: string;
+
+	/**
+	 * The netmask used by the address.
+
+	/**
+	 * The IP (v4 or v6) address of the gateway used by this IP address.
+	 */
+	gateway: string;
```

Comment:
gateway should be optional, not all IPs have an associated gateway.

Reply:
"Optional" means something very specific in TypeScript. It means that the property can be missing from an object, and I'd prefer allowing things to be `null`. Just want to be clear what you're suggesting.

Reply:
This is optional in the scope of database optionality.  I don't know what Typescript has to do with it.  In other words, this field should be nullable.

Reply:
Because that's a typescript block.

But the root of the matter was I just wasn't sure if he was suggesting that
```json
{
  "address": "0.0.0.0",
  "gateway": null,
  "serviceAddress": false
}
```
be acceptable or
```json
{
  "address": "0.0.0.0",
  "serviceAddress": false
}
```
be acceptable. And I prefer the former.

Reply:
Gateways are now allowed to be `null`